cmd/spire-agent/cli/run/run_test.go

```diff
@@ -858,6 +860,30 @@ func TestNewAgentConfig(t *testing.T) {
 				require.Nil(t, c)
 			},
 		},
+		{
+			msg: "warn_on_long_trust_domain",
+			input: func(c *Config) {
+				c.Agent.TrustDomain = strings.Repeat("a", 256)
+			},
+			logOptions: []log.Option{
```

Comment:
This is a very interesting use of `Cleanup()` :) However, it is using the wrong `*testing.T` instance for the assertions (the tests are run in a sub-test, and `t` here is for the main test. The test framework will panic if there is actually a problem.

In lieu of this I think it would be fairly trivial to instead add this hook down in the sub-test explicitly before calling NewAgentConfig and assert generally that the expected logs are output for each test case. It _may_ require adding some additional expected logs for existing test cases, but hopefully wouldn't be too bad?